src/components/bonds/individualBonds.tsx

```diff
@@ -0,0 +1,95 @@
+import Modal from 'react-bootstrap/Modal';
+import { useState } from 'react';
+import Plot from 'react-plotly.js';
+import { Data, Layout } from 'plotly.js';
+import { Bond, bondStatistics, bondStatisticsUtilities } from '../../bonds/bondsFunctionality';
+import "./individualBonds.css";
+
+export default function ProcessIndividualBonds(props: any) {
+  const bondStatisticsUtilitiesObject = new bondStatisticsUtilities();
+  const bondStatisticsObject = new bondStatistics();
+
+  const arrayOfAllPrices = bondStatisticsUtilitiesObject.returnArrayOfAllPrices(props.modalChartData);
+
+  const maxValue = bondStatisticsObject.findHighestNumber(arrayOfAllPrices);
+  const minValue = bondStatisticsObject.findLowestNumber(arrayOfAllPrices);
+
+  return (
+    <DisplayIndividualBonds
+      container={props.container}
+      modalChartData={props.modalChartData}
+      maxValue={maxValue}
+      minValue={minValue}
+    />
+  );
+}
+
+function DisplayIndividualBonds(props: any) {
+  const [showModal, setShowModal] = useState(false);
+
+  const handleDetailsButtonClick = () => setShowModal(true);
+  const handleHideModal = () => setShowModal(false);
+
+  if ((props as object) === undefined) return <div></div>;
+  //if (!props.hasOwnProperty("container")) return <div></div>;
+  //if (!props.hasOwnProperty("modalChartData")) return <div></div>;
```

Comment:
If these are no longer necessary, these commented out lines need to be cleaned up

Reply:
ok